### PR TITLE
Unary gRPC tracing example

### PR DESCRIPTION
**Pull request by davidmontoyago**
Simple tracing example for unary gRPC server and client using interceptors. Follows some of the patterns set by the HTTP example.

### CODE REVIEW

**Comment by jmacd** [comment]
@iredelmeier -- please review.

Note that #290 was closed because we expect imminently to introduce copies of the Datadog tracer plugins adapted to OpenTelemetry. I think we shouldn't turn away this example, but I do expect that we'll be able to use a standard middleware component in the near future.

File: example/grpc/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# gRPC Tracing Example
+
+Traces client and server calls via interceptors.
+
+### Compile .proto
```
Comment (iredelmeier) [inline example/grpc/README.md]: We should clarify that recompiling the protobufs is only necessary if you're changing them.

File: example/grpc/middleware/tracing/tracing.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	"go.opentelemetry.io/otel/api/key"
+	"go.opentelemetry.io/otel/api/trace"
+	"go.opentelemetry.io/otel/exporter/trace/stdout"
+	"go.opentelemetry.io/otel/global"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+func init() {
+	initOtelTracer()
+}
```
Comment (iredelmeier) [inline example/grpc/middleware/tracing/tracing.go]: Mind moving `initOtelTracer` out of `init`, e.g., into a helper function that can be called from the two `main` functions? My concern is that we don't want instrumentation to have initialization side effects like tracer or exporter construction, so doing so in an example might be misleading.

**Comment by iredelmeier** [comment]
@davidmontoyago I also want to clarify my understanding of your goal with the PR. The heart of it is about how to add the interceptor, right? In other words, if/when we switch to using the Datadog gRPC plugin, we would remove `middleware/tracing` and instead pull in the interceptors from the plugin?

File: example/grpc/middleware/tracing/tracing.go (hunk 2 above)
Comment (davidmontoyago) [reply example/grpc/middleware/tracing/tracing.go]: Makes sense. On it.

**Comment by davidmontoyago** [comment]
@iredelmeier that's correct. I was mainly intending to demo the use of interceptors but to also showcase a simple lib agnostic gRPC tracer that can be customized as it fits.

**Pull request merged**